build/push_container2ocp.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+#
+# Pushes local container image `service-assurance-operator` into OCP
+# * `oc` must already target OCP
+# * container image must already be built. See build.sh for more information
+#
+set -e
+source "$(dirname "$0")/metadata.sh"
+
+if [ "${IMAGE_BUILDER}" = "podman" ]; then
+    REG_EXTRAFLAGS="--tls-verify=false"
+fi
+
+${IMAGE_BUILDER} tag "${OPERATOR_NAME}:${IMAGE_TAG}" "${OCP_REGISTRY}/${OCP_PROJECT}/${OPERATOR_NAME}:${OCP_TAG}"
+${IMAGE_BUILDER} login ${REG_EXTRAFLAGS} -u 'openshift' -p "$(oc whoami -t)" "${OCP_REGISTRY}"
```

Comment:
On a non-CRC deployment (e.g. upshift / quicklab) this will fail because the login user is `kubeadmin`.

Reply:
Oh yeah? Been awhile since I've had one run to test. Note that the token we are presenting is for the kubeadmin user, it's just that all logins to the registry use the "openshift" user. Not sure why this would be different in CRC but I guess I'm not surprised. I'll try it in quicklab, though.

Reply:
I seem to have done it on both my CRC and quicklab tests with the following (although I should probably validate and test in CRC again):

```
REGISTRY=$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
TOKEN=$(oc whoami -t)
buildah login --tls-verify=false -u kubeadmin -p "${TOKEN}" "${REGISTRY}"
```

Not going to block on this. I can always test post-merge in upshift and file an issue and/or patch.

Reply:
Ah, neat, kubeadmin does work on CRC! I though that Matthias reported that it didn't.

```
[csibbitt@laptop service-assurance-operator (csibbitt-900-ci-scripts) ]$ ./build/push_container2ocp.sh 
Login Succeeded!
Getting image source signatures
Copying blob 4d253fad236e skipped: already exists
Copying blob ab6008cc8926 skipped: already exists
Copying blob acb86a0a1282 skipped: already exists
Copying blob 3b7c8c69c6ff skipped: already exists
Copying blob c630f5c3e169 skipped: already exists
Copying blob 4bd7408cc1c8 skipped: already exists
Copying blob 911fc1b18694 skipped: already exists
Copying blob 915be7a2cb0f skipped: already exists
Copying blob 1fd5c3a10c1a skipped: already exists
Copying blob 964b072de0ec skipped: already exists
Copying blob 34eed33aab83 skipped: already exists
Copying blob 3fc3a0fdbe1f skipped: already exists
Copying config 3fd22a2569 done
Writing manifest to image destination
Copying config 3fd22a2569 done
Writing manifest to image destination
Storing signatures
```

Still testing with quicklab. I'll break this out into an ENVVAR and set kubeadmin if it works on both.

Reply:
Quicklab registry is not exposed by default, but I found the instructions for exposing the registry here: https://docs.openshift.com/container-platform/4.3/registry/securing-exposing-registry.html

Curiously, the command they show for logging into the registry doesn't work, but user "openshift" does, as does "kubeadmin". (This was actually what Matthias reported before - we should look into where to log a bug for that)

```
$ echo $KUBECONFIG
/home/csibbitt/.kube/quicklab-stf001

$ podman login -u $(oc whoami) -p $(oc whoami -t) --tls-verify=false $(oc registry info)
Error: error logging into "default-route-openshift-image-registry.apps-crc.testing": invalid username/password

$ podman login -u openshift -p $(oc whoami -t) --tls-verify=false $(oc registry info)
Login Succeeded!

$ podman login -u kubeadmin -p $(oc whoami -t) --tls-verify=false $(oc registry info)
Login Succeeded!

$ oc whoami
kube:admin
```

I'll make the user an envvar anyways, but it doesn't seem like we'll need to change anything. Output of full test coming.

Reply:
Yea, I had that in my docs as well :)

```
# for use on upshift environments where image registry does not have a default route
# source:  https://docs.openshift.com/container-platform/4.2/registry/securing-exposing-registry.html
 
oc patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
```

Reply:
I should have noticed and called that out in my review